nixos/lib/test-driver/test_driver/__init__.py

```diff
@@ -63,6 +75,13 @@ def main() -> None:
         nargs="*",
         help="vlans to span by the driver",
     )
+    arg_parser.add_argument(
+        "output_directory",
```

Comment:
Having this as a positional argument breaks the current interface of allowing `nix build -f . nixosTests.foo && ./result/bin/nixos-test-driver`, would prefer a `--output-directory` flag with default value `$PWD`

Reply:
Do you mean `nix build -f . nixosTests.foo.driver && ./result/bin/nixos-test-driver`?  in that case one just needs to run `nix build -f . nixosTests.foo.driver && ./result/bin/nixos-test-driver . `
I made it positional, because I wanted to make use of the WriteableDir action which checks whether the out-dir exists. Since actions are not executed for default values I removed it. But I will make it optional as you suggest and check whether the out-dir is writeable right after parsing the arguments